modules/study/src/main/TreeBuilder.scala

```diff
@@ -2,54 +2,35 @@ package lila.study
 
 import chess.opening.*
 import chess.variant.Variant
-import lila.tree
+import lila.tree.{ Branch, Branches, Root }
 
 object TreeBuilder:
 
   private val initialStandardDests = chess.Game(chess.variant.Standard).situation.destinations
 
-  def apply(root: Node.Root, variant: Variant): tree.Root =
+  // DEBUG should be done in BSONHandler
+  def apply(root: Root, variant: Variant): Root =
     val dests =
       if (variant.standard && root.fen.isInitial) initialStandardDests
       else
         val sit = chess.Game(variant.some, root.fen.some).situation
         sit.playable(false) ?? sit.destinations
     makeRoot(root, variant).copy(dests = dests.some)
 
-  def toBranch(node: Node, variant: Variant): tree.Branch =
-    tree.Branch(
-      id = node.id,
-      ply = node.ply,
-      move = node.move,
-      fen = node.fen,
-      check = node.check,
-      shapes = node.shapes,
-      comments = node.comments,
-      gamebook = node.gamebook,
-      glyphs = node.glyphs,
-      clock = node.clock,
-      crazyData = node.crazyData,
-      eval = node.score.map(_.eval),
-      children = toBranches(node.children, variant),
+  // DEBUG should be done in BSONHandler
+  def toBranch(node: Branch, variant: Variant): Branch =
+    node.copy(
       opening = Variant.list.openingSensibleVariants(variant) ?? OpeningDb.findByEpdFen(node.fen),
-      forceVariation = node.forceVariation
+      children = toBranches(node.children, variant)
     )
```

Comment:
Opening only set when sending to the client, not type-safe.

See comment in BSONHandler.scala about it